datalad/distributed/create_sibling_gitlab.py

```diff
@@ -0,0 +1,598 @@
+                        "maybe use --existing=reconfigure",
+                        project, site,
+                        dremotes[remotename].get('url', None),
+                        site_project.get('ssh_url_to_repo', None)
+                    )
+                )
+                return
+            elif access == 'http' \
+                    and dremotes[remotename].get(
+                        'url', None) != site_project.get(
+                            # use False as a default so that there is a
+                            # mismatch, veen if both are missing
+                            'http_url_to_repo', False):
+                yield dict(
+                    res_kwargs,
+                    project_attributes=site_project,
+                    status='error',
+                    message=(
+                        "There is already a project at '%s' on site '%s', "
+                        "but HTTP access URL '%s' does not match '%s', "
+                        "maybe use --existing=reconfigure",
+                        project, site,
+                        dremotes[remotename].get('url', None),
+                        site_project.get('http_url_to_repo', None)
+                    )
+                )
+                return
+        yield dict(
+            res_kwargs,
+            project_attributes=site_project,
+            status='notneeded',
+            message=(
+                "There is already a project at '%s' on site '%s'",
+                project, site,
+            )
+        )
+
+    # first make sure that annex doesn't touch this one
+    # but respect any existing config
+    ignore_var = 'remote.{}.annex-ignore'.format(remotename)
+    if ignore_var not in ds.config:
+        ds.config.add(ignore_var, 'true', where='local')
+
+    for res in ds.siblings(
+            'configure',
+            name=remotename,
+            url=site_project['http_url_to_repo']
+            if access in ('http', 'ssh+http')
+            else site_project['ssh_url_to_repo'],
+            pushurl=site_project['ssh_url_to_repo']
+            if access in ('ssh', 'ssh+http')
+            else None,
+            recursive=False,
+            publish_depends=depends,
+            result_renderer='disabled',
+            return_type='generator'):
+        yield res
+
+
+class GitLabSite(object):
+    def __init__(self, site):
+        import gitlab
+        self.gitlab = gitlab
+        try:
+            self.site = gitlab.Gitlab.from_config(site)
+        except gitlab.config.GitlabDataError as e:
+            raise ValueError(
+                '{}, please configure access to this GitLab instance'.format(
+                    str(e)))
+
+    def get_project(self, path):
+        try:
+            return self.site.projects.get(path).attributes
+        except self.gitlab.GitlabGetError as e:
+            lgr.debug("Project with path '%s' does not yet exist at site '%s'",
+                      path, self.site.url)
```

Comment:
I would have made debug message more to the point -- "Failed to obtain project for path '%s'..." and include the actual exception (`exc_str`) into the the log message.  
I would assume that `GitlabGetError` might also be thrown for other reasons (connection? permissions?) and not only for a stated here assumption.  [Doc](https://python-gitlab.readthedocs.io/en/stable/api/gitlab.html) says
"GitlabGetError – If the server cannot perform the request" .
Code has lots of them and most relevant place of use:
```
gitlab/v4/objects.py=class ProjectJob(RESTObject, RefreshMixin):
--
gitlab/v4/objects.py-    @cli.register_custom_action("ProjectJob")
gitlab/v4/objects.py:    @exc.on_http_error(exc.GitlabGetError)
gitlab/v4/objects.py-    def artifacts(self, streamed=False, action=None, chunk_size=1024, **kwargs):
--
gitlab/v4/objects.py-            GitlabAuthenticationError: If authentication is not correct
gitlab/v4/objects.py:            GitlabGetError: If the artifacts could not be retrieved
gitlab/v4/objects.py-
--
gitlab/v4/objects.py-    @cli.register_custom_action("ProjectJob")
gitlab/v4/objects.py:    @exc.on_http_error(exc.GitlabGetError)
gitlab/v4/objects.py-    def artifact(self, path, streamed=False, action=None, chunk_size=1024, **kwargs):
--
gitlab/v4/objects.py-            GitlabAuthenticationError: If authentication is not correct
gitlab/v4/objects.py:            GitlabGetError: If the artifacts could not be retrieved
```
so smells that my connection (not authentication) assumption was on spot?